pkg/restapi/internal/common/did/common.go

```diff
@@ -116,9 +116,24 @@ func (o *CommonDID) CreateDID(keyType, signatureType, did, privateKey, keyID, pu
 		publicKeyID = keyID
 	}
 
+	didID, publicKeyID = o.replaceCanonicalDIDWithDomainDID(didID, publicKeyID)
+
 	return didID, publicKeyID, nil
 }
 
+func (o *CommonDID) replaceCanonicalDIDWithDomainDID(didID, publicKeyID string) (string, string) {
+	if strings.HasPrefix(didID, "did:trustbloc") {
+		split := strings.Split(didID, ":")
+		if len(split) == 4 {
+			domainDIDID := fmt.Sprintf("%s:%s:%s:%s", split[0], split[1], o.domain, split[3])
+
+			return domainDIDID, strings.ReplaceAll(publicKeyID, didID, domainDIDID)
```

Comment:
@troyronda So client will only keep track of alias(domain) did and always use that one for VC?

Reply:
@troyronda @sandrask yes we need to use did ID with domain for VC

Reply:
we will use Canonical ID with DID exchange